### PR TITLE
Added macro _fff_write_multiple

### PR DESCRIPTION
**Pull request by bamkrs**
The macro allows to write an array of elements to the fifo in a circular buffer manner.

### CODE REVIEW

**Comment by nqtronix** [comment]
Thanks you for this addition. I will test it in the next few days and merge it after I verified that it works as intended.

(sorry for getting back to you this late, I had notifications disabled for some reason and didn't notice)

**Comment by bamkrs** [comment]
Hey, no worries. I think I found a bug later on in my submission that I forgot to add to the PR. I'll be back at my machine tomorrow and update PR.

File: fifofast.h [hunk 1]
```diff
@@ -373,6 +373,30 @@ do{																\
...
+    }                                                           \
+    tocopy = _min(btw, _fff_mem_depth(_id) - _id.write);        \
+    memcpy(&_id.data[_id.write], (newdata), tocopy);            \
+    _id.level += tocopy;                                        \
+    _id.write = _fff_wrap(_id, (_id.write+tocopy));             \
+    btw -= tocopy;                                              \
+    if (btw > 0) {                                              \
+        memcpy(&_id.data[_id.write], (newdata)+tocopy, btw);    \
+        _id.write = btw;                                        \
+        _id.level += tocopy;                                    \
```
Comment (bamkrs) [inline fifofast.h]: This was the bug
```suggestion
        _id.level += btw;                                    \
```

**Review by bamkrs — COMMENTED** [review]
I compared it with my local copy and indeed fixed a bug

**Comment by nqtronix** [comment]
I've added the test case for buffer overflow and done some minor cosmetic charges. I didn't touch the code because everything worked as intended out of the box.

Thank you for your contribution!

**Pull request merged**